### DISCUSSION TITLE
Unexpected pane creation with run command in keybinding

### DISCUSSION THREAD

**Discussion by gunererd** [zellij-org/zellij#3810] [Q&A] [answered]
I’m trying to define a keybinding in Zellij to open a new tab and run lazygit in it. I’ve created a specific layout for lazygit, and I’m using the run command to achieve this. The setup works but with an unexpected issue: it creates a new tab with lazygit, but it also opens a new pane in the main tab.

```
// layouts/default.kdl
layout {
    default_tab_template {
        children
        pane size=1 borderless=false{
            plugin location="zellij:compact-bar"
        }
    }

    tab name="main"
}

keybinds {
    shared {
        bind "Alt g" { Run "zellij" "action" "new-tab" "-n" "git" "--layout" "lazygit";}
        bind "Alt q" { CloseFocus;} 
    }
}
```

```
// layouts/lazygit.kdl
layout {
    pane command="lazygit"
    pane size=1 borderless=false{
        plugin location="zellij:compact-bar"
    }
}
```

Here’s a video illustrating the issue. Why is it creating a new pane in the main tab, and how can I prevent this behavior? Any pointers would be appreciated!

https://github.com/user-attachments/assets/2aa2edcf-bb8a-4d2d-bea6-1f76c14ebed4

**Comment by imsnif** [comment, marked as answer]
I'd recommend using the `NewTab` keybind-action directly rather than invoking it from the CLI: https://zellij.dev/documentation/keybindings-possible-actions#newtab

When you invoke it from the CLI, you're essentially starting a new pane with the output of the above command `zellij action new-tab` whose sole purpose is to open a new tab. There are ways of getting this pane to close on exit, but I think if you use the above keybindable-action, this problem goes away.

**Reply by gunererd** [reply]
Ah it makes sense now. Also thanks for recommendation, worked perfectly.
